recipes/beyla-service-graph/graphgen/internal/query.go

```diff
@@ -54,20 +66,30 @@ func QueryPrometheus(ctx context.Context, client api.Client, queryWindow time.Du
 	graph := NewGraph()
 	for _, sample := range vec {
 		labels := sample.Metric
-		client := &Node{Ip: string(labels[clientIpKey])}
-		server := &Node{Ip: string(labels[serverIpKey]), Name: string(labels[serverPodKey])}
+		client := &Node{Ip: string(labels[clientIpKey]), Name: string(labels[clientKey])}
+		server := &Node{Ip: string(labels[serverIpKey]), Name: string(labels[serverKey])}
 		graph.AddEdge(client, server)
 	}
 	return graph, nil
 }
 
-func getQuery(queryWindow time.Duration) string {
+func getQuery(queryArgs QueryArgs) string {
+	filters := addFilter(nil, "cluster", queryArgs.Cluster)
+	filters = addFilter(filters, "namespace", queryArgs.Namespace)
 	return fmt.Sprintf(
 		`sum by (
-			k8s_pod_ip, client_address, k8s_pod_name
+			server, client_address, client, server_address
 		) (
-			rate(http_servicegraph_calls_total[%s])
+			rate(traces_service_graph_request_total{%s}[%s]) != 0
```

Comment:
The bug in #92 only happens for a short time after Beyla starts so I can filter out the zero valued rates